synapse/storage/event_federation.py

```diff
@@ -190,28 +190,41 @@ def get_latest_event_ids_and_hashes_in_room(self, room_id):
             room_id,
         )
 
-    def get_rooms_with_many_extremities(self, min_count, limit):
+    def get_rooms_with_many_extremities(self, min_count, limit, room_id_filter):
         """Get the top rooms with at least N extremities.
 
         Args:
             min_count (int): The minimum number of extremities
             limit (int): The maximum number of rooms to return.
+            room_id_filter (list[str]): List of room_ids to exclude from the look up
 
         Returns:
             Deferred[list]: At most `limit` room IDs that have at least
             `min_count` extremities, sorted by extremity count.
         """
 
         def _get_rooms_with_many_extremities_txn(txn):
+
+            where_clause = "1=1"
+            if room_id_filter:
+                where_clause = "room_id NOT IN (%s)" % (
+                    ",".join("?" for _ in room_id_filter),
+                )
+
             sql = """
                 SELECT room_id FROM event_forward_extremities
+                WHERE %s
                 GROUP BY room_id
                 HAVING count(*) > ?
                 ORDER BY count(*) DESC
                 LIMIT ?
-            """
+            """ % (
+                where_clause,
+            )
 
-            txn.execute(sql, (min_count, limit))
+            query_args = room_id_filter
```

Comment:
we shouldn't modify the input array:

```suggestion
            query_args = room_id_filter + [min_count, limit]
```